src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -365,39 +375,46 @@ private synchronized Integer availableNodeCapacity() throws IOException {
   @Override
   public boolean canProvision(Label label) {
     try {
-      getInstanceConfig(label);
+      getInstanceConfigurations(label);
       return true;
     } catch (NoConfigurationException nce) {
       return false;
     }
   }
 
-  /** Gets {@link InstanceConfiguration} that has the matching {@link Label}. */
-  public InstanceConfiguration getInstanceConfig(Label label) throws NoConfigurationException {
+  /** Gets all instances of {@link InstanceConfiguration} that has the matching {@link Label}. */
+  public List<InstanceConfiguration> getInstanceConfigurations(Label label)
+      throws NoConfigurationException {
     if (configurations == null) {
       throw new NoConfigurationException(
           String.format(
               "Cloud %s does not have any defined instance configurations.", this.getCloudName()));
     }
 
-    for (InstanceConfiguration configuration : configurations) {
-      if (configuration.getMode() == Node.Mode.NORMAL) {
-        if (label == null || label.matches(configuration.getLabelSet())) {
-          return configuration;
-        }
-      } else if (configuration.getMode() == Node.Mode.EXCLUSIVE) {
-        if (label != null && label.matches(configuration.getLabelSet())) {
-          return configuration;
-        }
-      }
+    List<InstanceConfiguration> configurations =
+        this.configurations.stream()
+            .filter(configuration -> matchesLabel(configuration, label))
+            .collect(Collectors.toList());
+
+    if (configurations.isEmpty()) {
+      throw new NoConfigurationException(
```

Comment:
Throwing exceptions like this is rather bad smell..
I didnt want to change it tho, but it should be fixed..